Examples/test-suite/preproc_predefined.h

```diff
@@ -25,3 +25,14 @@
 #  error Did not detect exactly one target-language-specific macro defined in the generated wrapper.
 # endif
 #endif
+
+/*
+ * This should not compile in when this bug is present
+ * https://github.com/swig/swig/issues/2525
+ */
+
+#define my_bad_attr __attribute__
+#define my_good_attr(x) __attribute__(x)
+
+int bad my_bad_attr((used));
+int good my_good_attr((used));
```

Comment:
You've added this to a testcase which is specifically testing SWIG's predefined macros, which this is not related to at all.  Where you had it before in `preproc_defined.i` seemed more suitable.